### PR TITLE
[LO-827] Fix/has-expr-string-formatting

### PR DESCRIPTION
**Pull request by JamesDanielOReilly**
- Fixed issue with error message not showing quotes if the output was a string
- Improved clarity of functions in utils.py
- Added tests to test for edge cases with newlines and strings in test_messaging.py
- Failed seven exercises, all working when tested manually. See: https://app.circleci.com/pipelines/github/datacamp-engineering/shared-images/380/workflows/997861c1-6b91-4d2b-8a86-504e2ee240ee/jobs/720

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 All notable changes to the pythonwhat project will be documented in this file. This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## 2.23.1
+
+- Fix string formatting in has_expr(). Strings will now have quotes in error messages. Leading and trailing whitespace is no longer removed.
```
Comment (hermansje) [inline CHANGELOG.md:7]: What causes the 'Leading and trailing whitespace is no longer removed.'?

Comment (hermansje) [reply CHANGELOG.md:7]: Ah, the formatting, I see it in the test as well.
